server/app/controllers/fusor/api/v21/subscriptions_controller.rb

```diff
@@ -15,10 +15,14 @@ class Api::V21::SubscriptionsController < Api::V2::BaseController
     skip_before_filter :check_content_type, :only => [:upload]
 
     def index
-      if params[:deployment_id]
+      if params[:deployment_id] && params[:source]
+          Rails.logger.warn "XXX filtering by deployment_id AND by source: #{params[:source]}"
+          @subscriptions = Fusor::Subscription.where(:deployment_id => params[:deployment_id],
+                                                     :source => params[:source])
+      elsif params[:deployment_id]
```

Comment:
@jmrodri, review this changes to the server controller. In short, if I passed `deployment_id` AND `source`, it only returned what was in the `deployment_id` and ignored the `source`